src/lib/stream.ts

```diff
@@ -0,0 +1,70 @@
+import { pipeline as pipelineSync } from 'node:stream'
+import { promisify } from 'node:util'
+
+import type { Handler, HandlerEvent, HandlerContext, StreamingHandler, StreamingResponse } from '../function/index.js'
+
+// Node v14 doesn't have node:stream/promises
+const pipeline = promisify(pipelineSync)
+
+declare global {
+  // eslint-disable-next-line @typescript-eslint/no-namespace
+  namespace awslambda {
+    function streamifyResponse(
+      handler: (event: HandlerEvent, responseStream: NodeJS.WritableStream, context: HandlerContext) => Promise<void>,
+    ): Handler
+
+    // eslint-disable-next-line @typescript-eslint/no-namespace
+    namespace HttpResponseStream {
+      function from(stream: NodeJS.WritableStream, metadata: Omit<StreamingResponse, 'body'>): NodeJS.WritableStream
+    }
+  }
+}
+
+/**
+ * Enables streaming responses. `body` accepts a `Readable` stream or Web stream.
+ *
+ * @example
+ * ```
+ * const { Readable } = require('stream');
+ *
+ * export const handler = stream(async (event, context) => {
+ *   const stream = Readable.from(Buffer.from(JSON.stringify(event)))
+ *   return {
+ *     statusCode: 200,
+ *     body: stream,
+ *   }
+ * })
+ * ```
+ *
+ * @example
+ * ```
+ * export const handler = stream(async (event, context) => {
+ *   const response = await fetch('https://api.openai.com/', { ... })
```

Comment:
nit: Maybe safer to use one of our own URLs in the example?

```suggestion
 *   const response = await fetch('https://api.netlify.com/api/v1/user', { ... })
```